app/models/subscription.rb

```diff
@@ -0,0 +1,35 @@
+class Subscription < ApplicationRecord
+  
+  belongs_to :owner, class_name: :User
+  belongs_to :plan
+  has_many :subscription_users
+  has_many :users, through: :subscription_users
+
+  validates :start_date, presence: true
+  validate :ensure_user_is_not_an_active_subscriber
+
+  before_create :set_expiry_date
+
+  after_create_commit :add_owner_to_users
+
+  def active_on?(date)
+    start_date <= date && expiry_date >= date
+  end
+
+  def remaining_days
+    (expiry_date - Date.today).to_i
```

Comment:
what should we return from this method if date is of past?

Reply:
Fixed, returning 0 in that case.